ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils_filtered_tenants.py

```diff
@@ -148,29 +149,36 @@ def _filtered_vrfs(self, tenant: dict) -> list[dict]:
-                        for rp_ip in get(
-                            rp_address,
-                            "rps",
-                            required=True,
-                            org_key=f"pim_rp_addresses.rps under VRF '{vrf['name']}' in Tenant '{tenant['name']}'",
-                        ):
-                            if rp_groups := get(rp_address, "groups"):
-                                rps.append({"address": rp_ip, "groups": rp_groups})
-                            else:
-                                rps.append({"address": rp_ip})
-                if rps:
-                    vrf["_pim_rp_addresses"] = rps
-
-                for evpn_peg in default(get(vrf, "evpn_l3_multicast.evpn_peg"), get(tenant, "evpn_l3_multicast.evpn_peg"), []):
-                    if self.shared_utils.hostname in evpn_peg.get("nodes", [self.shared_utils.hostname]) and rps:
-                        vrf["_evpn_l3_multicast_evpn_peg_transit"] = evpn_peg.get("transit")
-                        break
+            if self.shared_utils.vtep is True:
+                evpn_l3_multicast_enabled = default(get(vrf, "evpn_l3_multicast.enabled"), get(tenant, "evpn_l3_multicast.enabled"))
+                if evpn_l3_multicast_enabled is True and self._evpn_multicast is not True:
+                    raise AristaAvdError(
+                        f"'evpn_l3_multicast: true' under VRF {vrf['name']} or Tenant {tenant['name']}; this requires 'evpn_multicast' to also be set to true."
+                    )
+
+                if self._evpn_multicast:
+                    vrf["_evpn_l3_multicast_enabled"] = evpn_l3_multicast_enabled
+
+                    rps = []
+                    for rp_address in default(get(vrf, "pim_rp_addresses"), get(tenant, "pim_rp_addresses"), []):
+                        if self.shared_utils.hostname in get(rp_address, "nodes", default=[self.shared_utils.hostname]):
+                            for rp_ip in get(
+                                rp_address,
+                                "rps",
+                                required=True,
+                                org_key=f"pim_rp_addresses.rps under VRF '{vrf['name']}' in Tenant '{tenant['name']}'",
+                            ):
+                                if rp_groups := get(rp_address, "groups"):
+                                    rps.append({"address": rp_ip, "groups": rp_groups})
+                                else:
+                                    rps.append({"address": rp_ip})
+                    if rps:
+                        vrf["_pim_rp_addresses"] = rps
+
+                    for evpn_peg in default(get(vrf, "evpn_l3_multicast.evpn_peg"), get(tenant, "evpn_l3_multicast.evpn_peg"), []):
+                        if self.shared_utils.hostname in evpn_peg.get("nodes", [self.shared_utils.hostname]) and rps:
+                            vrf["_evpn_l3_multicast_evpn_peg_transit"] = evpn_peg.get("transit")
+                            break
```

Comment:
A large chunk of this will need to be refactored due to #3201 - but I'm happy to merge this one first and then I'll rework #3201.